src/UnliftIO/Retry.hs

```diff
@@ -0,0 +1,268 @@
+{-# LANGUAGE RankNTypes            #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  UnliftIO.Retry
+-- Copyright   :  Ozgun Ataman <ozgun.ataman@soostone.com>
+-- License     :  BSD3
+--
+-- Maintainer  :  Patrick Brisbin <pbrisbin@gmail.com>
+-- Stability   :  provisional
+--
+-- Unlifted "Control.Retry".
+--
+-- @since 0.9.3.0
+----------------------------------------------------------------------------
+
+
+module UnliftIO.Retry
```

Comment:
It might be a good idea to add some tests just for the stuff that this module adds rather than reexports. If possible, you might be able to extract generic versions of those tests and reuse them for the unliftio version.

Reply:
By adds I just mean the functions specifically defined here. I agree that it seems unlikely that the parts that this module adds specifically would cause problems since they're mostly just delegating to the existing instances. If it's easy to generalize the tests, go for it but if it looks like it's going to be a pain, I'm okay with just relying on the underlying retry tests.

Reply:
Check out the last commit and let me know if that strategy works.